bgpd/bgp_vty.c

```diff
@@ -7047,6 +7047,132 @@ ALIAS_HIDDEN(no_neighbor_filter_list, no_neighbor_filter_list_hidden_cmd,
 	     "Filter incoming routes\n"
 	     "Filter outgoing routes\n")
 
+/* Set advertise-map to the peer. */
+static int peer_advertise_map_set_vty(struct vty *vty, const char *ip_str,
+				      afi_t afi, safi_t safi,
+				      const char *advertise_str, bool condition,
+				      const char *condition_str)
+{
+	int ret = CMD_WARNING_CONFIG_FAILED;
+	struct peer *peer;
+	struct route_map *advertise_map;
+	struct route_map *condition_map;
+
+	peer = peer_and_group_lookup_vty(vty, ip_str);
+	if (!peer)
+		return ret;
+
+	condition_map = route_map_lookup_warn_noexist(vty, condition_str);
+	advertise_map = route_map_lookup_warn_noexist(vty, advertise_str);
+
+	ret = peer_advertise_map_set(peer, afi, safi, advertise_str,
+				     advertise_map, condition, condition_str,
+				     condition_map);
+
+	return bgp_vty_return(vty, ret);
+}
+
+static int peer_advertise_map_unset_vty(struct vty *vty, const char *ip_str,
+					afi_t afi, safi_t safi,
+					const char *advertise_str,
+					bool condition,
+					const char *condition_str)
+{
+	int ret = CMD_WARNING_CONFIG_FAILED;
+	struct peer *peer;
+	struct route_map *advertise_map;
+	struct route_map *condition_map;
+
+
+	peer = peer_and_group_lookup_vty(vty, ip_str);
+	if (!peer)
+		return ret;
+
+	condition_map = route_map_lookup_warn_noexist(vty, condition_str);
+	advertise_map = route_map_lookup_warn_noexist(vty, advertise_str);
+
+	ret = peer_advertise_map_unset(peer, afi, safi, advertise_str,
+				       advertise_map, condition, condition_str,
+				       condition_map);
+
+	return bgp_vty_return(vty, ret);
+}
+
+DEFUN (neighbor_advertise_map,
```

Comment:
I think we can handle simpler with DEFPY and bundled with `[no$no]` form as well.